rapidpro/src/main/java/org/motechproject/rapidpro/webservice/impl/ContactWebServiceImpl.java

```diff
@@ -81,43 +82,27 @@ public void deleteContactByUUID(UUID uuid) throws WebServiceException {
     public Contact getContactByUUID(UUID uuid) throws WebServiceException {
         try {
             LOGGER.debug(FINDING + uuid);
-            Map<String, String> params = new HashMap<>();
-            params.put(UUID, uuid.toString());
-            return getWithParams(params);
+            return getOneWithParams(UUID, uuid.toString(), CONTACTS_ENDPOINT, PAGED_RESPONSE_CONTACT_TYPE_REF);
         } catch (RapidProClientException | JsonUtilException e) {
-            throw new WebServiceException(ERROR_RETREIVING_UUID + uuid, e);
+            throw new WebServiceException(ERROR_RETRIEVING_UUID + uuid, e);
         }
     }
 
     @Override
     public Contact getContactByPhoneNumber(String phoneNumber) throws WebServiceException {
         try {
-            LOGGER.debug("Finding contact with phone number: " + phoneNumber);
-            Map<String, String> params = new HashMap<>();
-            params.put(PHONE, phoneNumber);
-            return getWithParams(params);
+            LOGGER.debug(FINDING_PHONE_NUMBER + phoneNumber);
+            return getOneWithParams(PHONE, phoneNumber, CONTACTS_ENDPOINT, PAGED_RESPONSE_CONTACT_TYPE_REF);
         } catch (RapidProClientException | JsonUtilException e) {
             throw new WebServiceException(ERROR_RETREIVING_PHONE + phoneNumber, e);
         }
     }
 
-    private Contact getWithParams(Map<String, String> params) throws WebServiceException, RapidProClientException, JsonUtilException {
-        InputStream response = client.executeGet(CONTACTS_ENDPOINT, MediaFormat.JSON, params);
-        PaginatedResponse<Contact> paginatedResponse = (PaginatedResponse<Contact>) JsonUtils.toObject(response, PAGED_RESPONSE_CONTACT_TYPE_REF);
-        if (paginatedResponse.getResults().size() == 0) {
-            return null;
-
-        } else if (paginatedResponse.getResults().size() == 1) {
-            return paginatedResponse.getResults().get(0);
-
-        } else {
-            throw new WebServiceException("Query Returned more than one contact.Parameters:  " + params.toString());
-        }
-    }
-
     private void closeInputStream(InputStream is) {
```

Comment:
IOUtils.closeQuietly does this exact thing, we can probably get rid of this method